File: latihan1_b/python/DataTim.py
class DataTim():

    #private atribute
    __namaTim = ""
    __negaraAsal = ""
    __tahunBerdiri = ""

    # constructor
    def __init__(self, nt="", na="", tb=0):
        self.__namaTim = nt
        self.__negaraAsal = na
        self.__tahunBerdiri = tb
        self.DaftarPemain = self.DataPemain()
    
    # Membuat set dan get untuk setiap variable
    def setNamaTim(self, nt):
        self.__namaTim = nt

    def getNamaTim(self):
        return self.__namaTim
    
    def setNegaraAsal(self, na):
        self.__negaraAsal = na

    def getNegaraAsal(self):
        return self.__negaraAsal

    def setTahunBerdiri(self, tb):
        self.__tahunBerdiri = tb
    
    def getTahunBerdiri(self):
        return self.__tahunBerdiri

    def setPemain(self, np):
        self.DaftarPemain.setNamaPemain(np)

    def setJumlahPemain(self, jp):
        self.DaftarPemain.setJumlah(jp)

    # Menampilkan data Tim termasuk nama pemain
    def showDataTim(self):
        print("\n>> " + str(self.__namaTim) + " <<")
        print("Negara Asal   : " + str(self.__negaraAsal))
        print("Tahun Berdiri : " + str(self.__tahunBerdiri))
        print("Daftar Pemain :")
        self.DaftarPemain.showDataPemain()
    
    # inner class
    class DataPemain():
        #private atribute
        namaPemain = []
        jumlah = 0

        # constructor
        def __init__(self):
            self.namaPemain = []
            self.jumlah = 0
        
        def setNamaPemain(self, np):
            self.namaPemain.append(np)

        def setJumlah(self, jp):
            self.jumlah = jp
        
        # menampilkan nama setiap pemain
        def showDataPemain(self):
            i = 0
            for i in range(self.jumlah):
                print("   " + str(i+1) + ". " + str(self.namaPemain[i]))

File: latihan1_b/php/DataTim.php
<?php

class DataTim
{
    private $namaTim = "";
    private $negaraAsal = "";
    private $tahunBerdiri = 0;
    private $pemain;
    private $jumlahPemain = 0;

    public function __construct($nt="", $na="", $tb=0, $jp=0)
    {
        $this->namaTim = $nt;
        $this->negaraAsal = $na;
        $this->tahunBerdiri = $tb;
        $this->jumlahPemain = $jp;
    }

    public function setNamaTim($nt = ""){
        $this->namaTim = $nt;
    }

    public function getNamaTim(){
        return $this->namaTim;
    }

    public function setNegaraAsal($na = ""){
        $this->negaraAsal = $na;
    }

    public function getNegaraAsal(){
        return $this->negaraAsal;
    }

    public function setTahunBerdiri($tb){
        $this->tahunBerdiri = $tb;
    }

    public function getTahunBerdiri(){
        return $this->tahunBerdiri;
    }

    public function setJumlahPemain($jp){
        $this->jumlahPemain = $jp;
    }

    public function getJumlahPemain(){
        return $this->jumlahPemain;
    }

    public function setPemain($namaPemain){
        $this->pemain[] = new DataPemain($namaPemain);
    }

    public function getPemain($i){
        return $this->pemain[$i]->namaPemain;
    }
}

class DataPemain
{
    public $namaPemain = "";

    public function __construct($np="")
    {
        $this->namaPemain = $np;
    }
}